docs/getting-started-guides/docker-multinode/functions.sh

```diff
@@ -0,0 +1,343 @@
+      --net=host gcr.io/google_containers/etcd:${ETCD_VERSION} \
+      etcdctl \
+      set /coreos.com/network/config \
+      '{ "Network": "10.1.0.0/16", "Backend": {"Type": "vxlan"}}'
+}
+
+etcd_join() {
+  if [ ! -e "/etc/etcd-cluster.conf" ]; then
+    echo "Joining etcd cluster"
+    etcdctl --endpoint http://$MASTER_IP:4001 member add $NODE_NAME http://$CURRENT_IP:2380 | tail -n 3 > /etc/etcd-cluster.conf
+  fi
+  # Load configuration
+  source /etc/etcd-cluster.conf
+  echo "Will load etcd with"
+  echo " - ETCD_NAME=$ETCD_NAME"
+  echo " - ETCD_INITIAL_CLUSTER=$ETCD_INITIAL_CLUSTER"
+  echo " - ETCD_INITIAL_CLUSTER_STATE=$ETCD_INITIAL_CLUSTER_STATE"
+  docker-bootstrap run --restart=always --net=host --name=etcd -d gcr.io/google_containers/etcd:${ETCD_VERSION} /usr/local/bin/etcd \
+           --listen-client-urls=http://0.0.0.0:2379,http://0.0.0.0:4001 \
+           --advertise-client-urls=http://$CURRENT_IP:2379,http://$CURRENT_IP:4001 \
+           --data-dir=/var/etcd/data \
+           --listen-peer-urls http://0.0.0.0:2380 \
+           --name $NODE_NAME \
+           --initial-advertise-peer-urls http://$CURRENT_IP:2380 \
+           --initial-cluster-token $ETCD_NAME --initial-cluster $ETCD_INITIAL_CLUSTER --initial-cluster-state $ETCD_INITIAL_CLUSTER_STATE
+}
+
+flannel() {
+  if docker_bootstrap_run flannel; then
+    echo "Flannel is already running"
+    return 0
+  fi
+  echo "Launching flannel"
+  flannelCID=$(docker -H unix:///var/run/docker-bootstrap.sock run \
+      --restart=always \
+      -d \
+      --net=host \
+      --name=flannel \
+      --privileged \
+      -v /dev/net:/dev/net \
+      quay.io/coreos/flannel:${FLANNEL_VERSION} \
+      /opt/bin/flanneld \
+      --ip-masq \
+      -iface="${FLANNEL_IF}")
+
+      sleep 8
+      # Copy flannel env out and source it on the host
+      docker -H unix:///var/run/docker-bootstrap.sock \
+          cp ${flannelCID}:/run/flannel/subnet.env .
+      source subnet.env
+      detect_lsb
+      # Configure docker net settings, then restart it
+      case "${lsb_dist}" in
+          amzn)
+              DOCKER_CONF="/etc/sysconfig/docker"
+              echo "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | sudo tee -a ${DOCKER_CONF}
+              ifconfig docker0 down
+              yum -y -q install bridge-utils && brctl delbr docker0 && service docker restart
+              ;;
+          centos)
+              DOCKER_CONF="/etc/sysconfig/docker"
+              echo "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | sudo tee -a ${DOCKER_CONF}
+              if ! command_exists ifconfig; then
+                  yum -y -q install net-tools
+              fi
+              ifconfig docker0 down
+              yum -y -q install bridge-utils && brctl delbr docker0 && systemctl restart docker
+              ;;
+          ubuntu|debian)
+              DOCKER_CONF="/etc/default/docker"
```

Comment:
I dont remember having change this, but it might have evolved without me noticing